nir/src/main/scala/scala/scalanative/nir/Defns.scala

```diff
@@ -2,33 +2,78 @@ package scala.scalanative
 package nir
 import scala.scalanative.nir.Defn.Define
 
+/** A definition in NIR.
+ *
+ *  Programs in NIR are represented as a sequence of definitions denoting types,
+ *  methods and fields. Definitions fall into two categories:
+ *
+ *    - Top-level definitions: these represent classes, modules, or traits.
+ *      Classes and modules inherit from a single parent with the exception of
+ *      `java.lang.Object`, which sits at the top of the hierarchy. They may
+ *      additionally implement traits.
+ *    - Member definitions: these represent fields or methods.
```

Comment:
Because we also have `Const` declarations, I am not sure this comment is true. Is a constant always describing a member?

Reply:
It depends from the stage.
The `Defn.Var` in the form of `Defn.Var(_, Global.Member(_, Sig.Field(_),_,_)` exists only until lowering, after which the information about the field exists only in the `MemoryLayour` or `FieldLayout` of given type. I don't think we do define `Defn.Const` for `Sig.Field` at all. 

All the remaining `Defn.Var` and `Defn.Const` are basically a global variables/constants. At this point they would typically would be defined a `Member` with `Sig.Extern` or `Sig.Generated`. They can be used for example for keeping the stack bottom of the main thread or other information required at runtime. `Defn.Const` for example is used to create instances of strings or create the instances of `java.lang.Class[T]` (actually I think now we use Var for that)